realm/realm-library/src/main/java/io/realm/RealmCache.java

```diff
@@ -196,11 +193,6 @@ public void run() {
     // doCreateRealmOrGetFromCache is called with globalCount == 0.
     private RealmConfiguration configuration;
 
-    // Column indices are cached to speed up opening typed Realm. If a Realm instance is created in one thread, creating
-    // Realm instances in other threads doesn't have to initialize the column indices again.
-    private static final int MAX_ENTRIES_IN_TYPED_COLUMN_INDICES_ARRAY = 4;
-    private final ColumnIndices[] typedColumnIndicesArray = new ColumnIndices[MAX_ENTRIES_IN_TYPED_COLUMN_INDICES_ARRAY];
```

Comment:
The benchmark results of insertion actually shows it. Although it is running in single thread mode, the original implementation cached the column indices before insertion, current implementation only caches it when insertion called on the object. The benchmark doesn't show noticeable difference.